backend/src/main/java/eventlistener/service/event/EventService.java

```diff
@@ -1,56 +1,101 @@
+    public EventService(EventRepo eventRepo, EventMapper eventMapper) {
         this.eventRepo = eventRepo;
-        this.mapper = mapper;
+        this.eventMapper = eventMapper;
 
     }
 
-    public List<ResponseEvent> getAllEvents() {
-        return mapper.mapResponseEvents(eventRepo.findAll());
+    public List<Event> getAllEvents() {
+        return eventRepo.findAll();
     }
 
     public Event addEvent(Event eventToAdd) {
         return eventRepo.save(eventToAdd);
     }
 
-    public List<Event> getAllEventsFromUser(String userId) {
-        return eventRepo.findAllByNotificationUserContains(userId);
+    public List<Event> getAllEventsFromUser(NotificationUser user) {
+        return eventRepo.findAllByNotificationUserContains(user);
+    }
+
+    public List<Event> getAllEventsExcludeUser(NotificationUser users) {
+        return eventRepo.findAllByNotificationUserNotContaining(users);
     }
 
 
-    public boolean eventsExist(List<String> listenEvents) {
+    public boolean eventsExist(List<Long> listenEvents) {
         List<Event> listOfEvents = (List<Event>) eventRepo.findAllById(listenEvents);
         return listOfEvents.size() >= listenEvents.size();
     }
 
-    public void addUserToEvents(List<String> listenEvents, NotificationUser notificationUser) {
-        for (String eventId : listenEvents) {
+    public void addUserToEvents(List<Long> listenEvents, NotificationUser user) {
+        for (Long eventId : listenEvents) {
             Event eventToEdit = getEventById(eventId);
-            List<String> userList = eventToEdit.getNotificationUser();
-            userList.add(notificationUser.getId());
-            eventRepo.save(eventToEdit);
+            if(!userExistInEvent(user.getId(), eventToEdit)){
+                List<NotificationUser> userList = eventToEdit.getNotificationUser();
+                userList.add(user);
+                eventRepo.save(eventToEdit);
+            }
+        }
+    }
 
+    public List<Event> setNotificationUserToEvents(NotificationUser user, List<Event> eventsToSet){
+        clearAllEventLinks(user);
+        List<Event> newUserEvents = new ArrayList<>();
+        for (Event event : eventsToSet) {
+            newUserEvents.add(addUserToEvent(event.getId(), user));
         }
+        return newUserEvents;
+    }
+
+    public Event addUserToEvent(long eventId, NotificationUser userId){
+        Event eventToEdit = eventRepo.findById(eventId).orElseThrow(() -> new NoSuchElementException("No event found with ID: "+eventId));
+        eventToEdit.getNotificationUser().add(userId);
+        return eventRepo.save(eventToEdit);
     }
 
-    public Event getEventById(String eventId){
+    private void clearAllEventLinks(NotificationUser user){
+        List<Event> userEvents = eventRepo.findAllByNotificationUserContains(user);
+        for (Event userEvent : userEvents) {
+            userEvent.getNotificationUser().remove(user);
```

Comment:
does that work like this? don't you have to mutate the list and then use the setter t ochange it?